components/ads/Advertisement.vue

```diff
@@ -109,6 +109,7 @@ export default {
     // Register hook on resize
     window.addEventListener('resize', this.handleWindowResize)
     this.isDark = this.$colorMode.value !== 'light'
+    window.addEventListener('darkMode', this.handleDarkModeChange)
```

Comment:
The main reason why the ads weren't reactive to the dark mode change was to avoid broken tracking pixels in the process due to issues with Vue & Ethical Ads. This may need more testing than the rest